### PR TITLE
Improvements of Trie Structure

### PR DESCRIPTION
**Pull request by lucidfrontier45**
This PR improved `Trie` with the following two points.

1. fix `TrieIterator` to correctly handle non-ascii characters
2. add `find_prefixes` function which is equivalent to `prefixes` method of https://github.com/pytries/marisa-trie

### CODE REVIEW

File: src/trie.jl [hunk 1]
```diff
@@ -111,14 +111,27 @@ end
...
         t = t.children[it.str[i]]
-        return (t, (t, i + 1))
+        return (t, (t, nextind(it.str, i)))
     end
 end
 
 partial_path(t::Trie, str::AbstractString) = TrieIterator(t, str)
 Base.IteratorSize(::Type{TrieIterator}) = Base.SizeUnknown()
+
+function find_prefixes(t::Trie, str::AbstractString)
```
Comment (oxinabox) [inline src/trie.jl:126]: this requires a docstring

File: docs/src/trie.md [hunk 2]
```diff
@@ -31,3 +31,10 @@ given string, use:
 ```julia
 seen_prefix(t::Trie, str) = any(v -> v.is_key, partial_path(t, str))
 ```
+
+`find_prefixes` can be used to find all keys which are prefixes of the given string.
+
+```julia
+t = Trie(["A", "ABC", "ABCD", "BCE"])
+find_prefixes(t, "ABCDE") # "A", "ABC"
```
Comment (oxinabox) [inline docs/src/trie.md:39]: why does this not return `ABCD` as well?

**Comment by oxinabox** [comment]
Thanks.

Note that we are currently feature frozen pending #479 
and will not be making a 0.19 release with new features until that is done.
(something that I personally do not have time to do, so it is waiting for people who do)


The _fix TrieIterator to correctly handle non-ascii characters_ change is a good bug fix,  which we can backpoint to the 0.18.x line.
`find_prefixes` is not.

File: src/trie.jl (hunk 1 above)
Comment (lucidfrontier45) [reply src/trie.jl:142]: added in the latest commit

File: docs/src/trie.md (hunk 2 above)
Comment (lucidfrontier45) [reply docs/src/trie.md]: That's my mistake. I fixed it.

**Comment by lucidfrontier45** [comment]
@oxinabox 

Thank you for your comment. I updated my commits according to your suggestion.
I also understood the release schedule. I don't have any problem with it.

**Review by oxinabox — APPROVED** [review]

**Pull request merged**